mypy/semanal.py

```diff
@@ -817,6 +817,12 @@ def analyze_namedtuple_classdef(self, defn: ClassDef) -> bool:
                         items, types, default_items = self.check_namedtuple_classdef(defn)
                         node.node = self.build_namedtuple_typeinfo(
                             defn.name, items, types, default_items)
+                        # We only really need the assignments in the body to be type checked later;
+                        # attempting to type check methods may lead to crashes because NamedTuples
+                        # do not have a fully functional TypeInfo.
+                        # TODO remove this hack and add full support for NamedTuple methods
+                        defn.defs.body = [stmt for stmt in defn.defs.body
```

Comment:
This is safe because statements within a NamedTuple class fall in three groups (see check_namedtuple_classdef below):
1. assignments, which we still type check
2. pass and Ellipsis, which don't do anything and which we ignore here
3. Other statements, which produce an error on line 841/847 below and now are ignored here.